code/game/objects/items/robot/items/tools.dm

```diff
@@ -265,20 +275,28 @@
 			return
 		if(!user.Adjacent(target))
 			return
-		stored_crates += picked_crate
+		if(!owner.cell.use(cell_usage))
+			to_chat(user, span_warning("Not enough power."))
+			return
+		if(istype(target, /obj/structure/closet))
+			var/obj/structure/closet/c = target
+			if(c.opened)
+				balloon_alert(host, "close it first!")
+				return
```

Comment:
bu ikisinin do_after'dan önce de olması lazım özellikle "close it first"ün
yaparken de
```
var/obj/structure/closet/C = target
if(istype(C))
```
şeklinde yaparsan iyi olur daha az duplicate için

Reply:
bunun dışında sorun yok mergelenebilir